File: java/JavaFileutil/src/JavaFileutil/AbstractCommands/AbstractUnaryFileCommand.java
package JavaFileutil.AbstractCommands;

import JavaFileutil.Validators.ArgumentCountValidator;

import java.io.File;
import java.util.List;

public abstract class AbstractUnaryFileCommand extends AbstractFileCommand
{
    protected static final String ArgumentCountMismatchMessage = "the command takes EXACTLY 1 argument!";

    protected File file;

    private final void SetupValidation()
    {
        Validation.AddValidator(new ArgumentCountValidator(Arguments, (short)1, ArgumentCountMismatchMessage));
    }

    public AbstractUnaryFileCommand(List<String> args)
    {
        super(args);
        SetupValidation();

        file = (Arguments.size() >= 1 && !Arguments.get(0).isEmpty()) ? new File(Arguments.get(0)) : null;
    }
}

File: python/Abstractions/AbstractUnaryFileCommand.py
from .AbstractCommand import *
from Validators.ArgumentCountValidator import *
from Validators.DirectoryNonExistenceValidator import *
from pathlib import Path


class AbstractUnaryFileCommand(AbstractCommand):
    @staticmethod
    def _argument_count_mismatch_message():
        return "the command takes EXACTLY 1 argument!"

    @staticmethod
    def _argument_is_folder_message():
        return "this is a file command, the argument can't be a folder!"

    def _unary_validation(self):
        self._validation.add_validator(DirectoryNonExistenceValidator(self._file, self._argument_is_folder_message()))
        self._validation.add_validator(ArgumentCountValidator(self.arguments, 1, self._argument_count_mismatch_message()))

    def __init__(self, args):
        super().__init__(args)
        self._file = Path(self.arguments[0]) if self.arguments.__len__() > 0 else Path("")
        self._unary_validation()

    @abstractmethod
    def execute(self):
        pass

